dvc/command/plots.py

```diff
@@ -198,6 +210,16 @@ def add_parser(subparsers, parent_parser):
     )
     plots_modify_parser.set_defaults(func=CmdPlotsModify)
 
+    SHOW_HELP = "Dump plots templates."
+    plots_dump_parser = plots_subparsers.add_parser(
```

Comment:
Does it make sense to be able to specify a different path to dump to? Maybe I am worried about overwriting what I have now, or I've made modifications to the defaults.

Also, can I dump just one template? This seems less necessary, but it might be nice to not have to generate all the templates if I only want one, especially if the number of templates grows.

Reply:
I think it makes sense to extend the functionality of templates, though it raises another questions.
1. If we want to specify particular template, do we want any means of obtaining templates available for this particular repo? I think so, that would make the user experience much better than circling between terminal and docs.
2. If we agree in 1, should `templates` have subcommands? eg (`write` and `list`)? That would make sense and be unified with our other commands.

Reply:
When dumping `targets`, I was thinking of the default templates, not any file, since otherwise it's just copying a file, right? Is there a use case you had in mind? I see that it works to do something like `dvc plots template linear`, but it's not that clear from the help description that `targets` doesn't have to specify existing files. If I already have `linear.json` in `.dvc/plots`, will `dvc templates linear` dump the internal version or the one on file?

Having a way to list templates seems great. Maybe `--list` can be an option to avoid subcommands?

Reply:
I think we should throw an exception if we are trying to dump a template that is different from the existing one under the given path. Do you want a new one? Fine, remove the old one manually. No chance of accidental removal of custom changes.

Reply:
> Is there a use case you had in mind?

Not exactly, just wanted to clarify

Reply:
So is this the expected workflow to upgrade outdated templates in `.dvc/plots`?

1. `rm -rf .dvc/plots`
2. `dvc plots templates`

Reply:
Yes

Reply:
> If I already have `linear.json` in `.dvc/plots`, will `dvc templates linear` dump the internal version or the one on file?

So this will fail because it will overwrite `.dvc/plots/linear.json`, correct? What about `dvc templates linear -o new_linear.json`? Which template version will it dump?

Reply:
> So this will fail because it will overwrite .dvc/plots/linear.json, correct?

It will fail only if the content of the template differs.

> dvc templates linear -o new_linear.json?

About that, another question: do we want to specify particular file? Currently it is implemented that out is dir to dump templates to (to be able to dump few at once). But I guess it makes more sense to specify particular templates and particular paths.

> Which template version will it dump?

The one that is loaded from DVC code, so the newest one.

Reply:
> It will fail only if the content of the template differs.

Sounds good, although it also seems like more work for you to check if the contents are the same. It also seems fine to me to fail anytime a file would be overwritten.

> do we want to specify particular file?

No, sorry, I got confused. No need to specify a particular file.

> > Which template version will it dump?
> 
> The one that is loaded from DVC code, so the newest one.

So will `dvc plots templates` always operate on the built-in templates? Maybe we should specify in the help description if so.

Reply:
> No, sorry, I got confused. No need to specify a particular file.

I probably should explaining it in option description 

> So will dvc plots templates always operate on the built-in templates? Maybe we should specify in the help description if so.

Will do